internal/store/model/resource.go

```diff
@@ -32,11 +32,11 @@ type Resource struct {
 
 	Annotations pq.StringArray `gorm:"type:text[]"`
 
-	Generation *int64
-
-	CreatedAt time.Time
-	UpdatedAt time.Time
-	DeletedAt gorm.DeletedAt `gorm:"index"`
+	Generation      *int64
+	ResourceVersion *int64
```

Comment:
Why a pointer if it always exists?

Reply:
When the existing and the new resourceVersion(s) are compared, if it is nil (when it comes from the API) we ignore the version check.
In addition, in general I prefer to detect cases when it was not set for some reason.